### ISSUE TITLE
metric should have TP FP TN FN Precision Recall F1 for both macro and micro versions

### ISSUE THREAD

**Issue by szha** [apache/incubator-mxnet#9588]
Related to #9587. The current logic only supports what's called "macro F1" logic where we take the mean of F1 scores for mini-batches. "micro F1" should also be supported by keeping the tp, fp, fn counts across batches.
Note that these metrics share the counter logics so we should be mindful when doing oo design.

https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/metric.py

**Comment by szha** [comment]
We may refer to the "average" option in [sklearn.metrics.f1_score](http://scikit-learn.org/stable/modules/generated/sklearn.metrics.f1_score.html#sklearn.metrics.f1_score).

**Comment by ThomasDelteil** [comment]
@szha looks like #9777 address this ? If so please consider closing.

**Comment by szha** [comment]
Indeed
